### PR TITLE
subscriber: remove deprecated items and notices

### PR DESCRIPTION
**Pull request by davidbarsky**
This PR removes the deprecated items, notices and aliases in preparation of releasing tracing-subscriber 0.2.0-alpha.1.

### CODE REVIEW

File: tracing-subscriber/src/fmt/mod.rs [hunk 1]
```diff
@@ -608,18 +579,6 @@ impl<N, E, F, W> SubscriberBuilder<N, E, F, W> {
...
-    #[deprecated(since = "0.2.0", note = "renamed to `event_format`.")]
-    pub fn on_event<E2>(self, fmt_event: E2) -> SubscriberBuilder<N, E2, F, W>
-    where
-        E2: FormatEvent<Registry, N> + 'static,
-        N: for<'writer> FormatFields<'writer> + 'static,
-        W: MakeWriter + 'static,
-    {
-        self.event_format(fmt_event)
-    }
-
```
Comment (hawkw) [inline tracing-subscriber/src/fmt/mod.rs]: this should not have been removed; it was renamed in 0.2.0 so there hasn't yet been a published version where this method was deprecated

File: tracing-subscriber/src/lib.rs [hunk 2]
```diff
@@ -107,7 +107,7 @@ pub(crate) mod thread;
 
 #[cfg(feature = "env-filter")]
 #[allow(deprecated)]
```
Comment (hawkw) [inline tracing-subscriber/src/lib.rs]: should be able to remove this `allow(deprecated)`

Comment (hawkw) [reply tracing-subscriber/src/lib.rs]: (we should probably check for any other `allow(deprecated)`s?)

File: tracing-subscriber/src/fmt/mod.rs (hunk 1 above)
Comment (davidbarsky) [reply tracing-subscriber/src/fmt/mod.rs]: ah, fixed. thanks!

File: tracing-subscriber/src/lib.rs (hunk 2 above)
Comment (davidbarsky) [reply tracing-subscriber/src/lib.rs]: fixed!

File: tracing-subscriber/benches/fmt.rs [hunk 3]
```diff
@@ -57,7 +57,7 @@ fn bench_new_span(c: &mut Criterion) {
                         })
                         .thread(move || {
                             for n in 0..i {
-                                let _span = tracing::info_span!("span", i);
+                                let _span = tracing::info_span!("span", n);
```
Comment (hawkw) [inline tracing-subscriber/benches/fmt.rs]: unrelated change?

Comment (davidbarsky) [reply tracing-subscriber/benches/fmt.rs]: I mentioned that I'd fix this as well, but I can remove it.

Comment (hawkw) [reply tracing-subscriber/benches/fmt.rs]: that's fine, i forgot.
